src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/DynamicObjectResolverTests.cs

```diff
@@ -65,6 +66,182 @@ private void PrivateMembersInBaseClass_Helper(MessagePackSerializerOptions optio
+            writer.WriteMapHeader(1);
+            writer.Write(nameof(DefaultValueStringKeyStructWithExplicitConstructor.Prop1));
+            writer.Write(-1);
+            writer.Flush();
+
+            var instance = MessagePackSerializer.Deserialize<DefaultValueStringKeyStructWithExplicitConstructor>(seq);
+            Assert.Equal(-1, instance.Prop1);
+            Assert.Equal(DefaultValueStringKeyStructWithExplicitConstructor.Prop2Constant, instance.Prop2);
+        }
+
+        [Fact]
+        public void DefaultValueIntKeyClassWithoutExplicitConstructorTest()
+        {
+            var seq = new Sequence<byte>();
+            var writer = new MessagePackWriter(seq);
+            writer.WriteArrayHeader(0);
+            writer.Flush();
+
+            var instance = MessagePackSerializer.Deserialize<DefaultValueIntKeyClassWithoutExplicitConstructor>(seq);
+            Assert.Equal(DefaultValueIntKeyClassWithoutExplicitConstructor.Prop1Constant, instance.Prop1);
+            Assert.Equal(DefaultValueIntKeyClassWithoutExplicitConstructor.Prop2Constant, instance.Prop2);
+        }
+
+        [Fact]
+        public void DefaultValueIntKeyClassWithExplicitConstructorTest()
+        {
+            var seq = new Sequence<byte>();
+            var writer = new MessagePackWriter(seq);
+            writer.WriteArrayHeader(1);
+            writer.Write(-1);
+            writer.Flush();
+
+            var instance = MessagePackSerializer.Deserialize<DefaultValueIntKeyClassWithExplicitConstructor>(seq);
+            Assert.Equal(-1, instance.Prop1);
+            Assert.Equal(DefaultValueIntKeyClassWithExplicitConstructor.Prop2Constant, instance.Prop2);
+        }
+
+        [Fact]
+        public void DefaultValueIntKeyStructWithExplicitConstructorTest()
+        {
+            var seq = new Sequence<byte>();
+            var writer = new MessagePackWriter(seq);
+            writer.WriteArrayHeader(1);
+            writer.Write(-1);
+            writer.Flush();
+
+            var instance = MessagePackSerializer.Deserialize<DefaultValueIntKeyStructWithExplicitConstructor>(seq);
+            Assert.Equal(-1, instance.Prop1);
+            Assert.Equal(DefaultValueIntKeyStructWithExplicitConstructor.Prop2Constant, instance.Prop2);
+        }
+
+        [MessagePackObject(true)]
+        public class DefaultValueStringKeyClassWithoutExplicitConstructor
+        {
+            public const int Prop1Constant = 11;
+            public const int Prop2Constant = 45;
+
+            public int Prop1 { get; set; } = Prop1Constant;
+
+            public int Prop2 { get; set; } = Prop2Constant;
+        }
+
+        [MessagePackObject(true)]
+        public class DefaultValueStringKeyClassWithExplicitConstructor
+        {
+            public const int Prop2Constant = 1419;
+
+            public int Prop1 { get; set; }
+
+            public int Prop2 { get; set; }
+
+            public DefaultValueStringKeyClassWithExplicitConstructor(int prop1)
```

Comment:
In such a configuration, is it *possible* that `Prop2` will ever have a non-default value? Will MessagePack's dynamic resolver call the `Prop2` setter after invoking the constructor?

Reply:
Yes, it will call `Prop2` setter after constructor when the input byte[] has the property.
I add test codes.